src/hidimstat/marginal/partial_dependence_plot.py

```diff
@@ -0,0 +1,807 @@
+from collections.abc import Iterable
+from copy import deepcopy
+from tqdm import tqdm
+import warnings
+
+from joblib import Parallel, delayed
+import numpy as np
+from scipy import sparse
+from scipy.stats.mstats import mquantiles
+from sklearn.base import is_classifier, is_regressor
+from sklearn.utils import _safe_indexing, check_array, check_random_state
+from sklearn.utils._indexing import (
+    _determine_key_type,
+    _get_column_indices,
+    _safe_assign,
+)
+from sklearn.utils._response import _get_response_values
+from sklearn.utils.validation import _check_sample_weight, check_is_fitted
+from sklearn.inspection._pd_utils import _check_feature_names, _get_feature_index
+from sklearn.inspection._partial_dependence import _grid_from_X
+
+from hidimstat.base_variable_importance import (
+    BaseVariableImportance,
+)
+
+
+def _grid_from_X(
+    X,
+    percentiles,
+    is_categorical,
+    grid_resolution,
+    custom_values,
+    resolution_statistique=False,
+):
+    """
+    Generate a grid of points based on the percentiles of X.
+    Parameters
+    ----------
+    X : array-like of shape (n_samples, n_target_features)
+        The data from which to generate the grid.
+    percentiles : tuple of float (p1, p2)
+        The lower and upper percentiles to use for the grid boundaries.
+        Must satisfy 0 <= p1 < p2 <= 1.
+    is_categorical : list of bool
+        For each feature, indicates whether it is categorical or not.
+        For categorical features, unique values are used instead of percentiles.
+    grid_resolution : int
+        Number of equally spaced points for the grid.
+        Must be greater than 1.
+    custom_values : dict
+        Mapping from column index of X to array-like of custom values
+        to use for that feature instead of the generated grid.
+    resolution_statistique : bool, default=False
+        If True, uses quantiles for grid points instead of equally spaced points
+        between percentile boundaries.
```

Comment:
Can we drop a line here for readability ?

Reply:
done